src/Expr.cpp

```diff
@@ -0,0 +1,12 @@
+#include "Expr.h"
+#include "IR.h"
+
+namespace Halide {
+// FIXME: This feels like a horrible hack. FloatImm should be able
```

Comment:
@abadams Although "Use a cast of a float" is the least invasive change I'm leaning (and I'm going to try)  ``Making FloatImm more general to serve a rare use-case, affecting every use of it`` because I need to know where it is going to be used (e.g. constant folding needs to be done differently) at the same time I could also add support for representing double immediates.